finetune/encoding.py

```diff
@@ -157,45 +180,82 @@ def encode_for_classification(self, texts, max_length, verbose=True):
+        whilst keeping the overall sequence length within max_length tokens. It also adds the 3 special tokens. Start,
+         Classify and Delimiter.
+        :param encoded: Lists of shape [sequences, batch, num_tokens]
+        :param max_length: Int representing the max length of a single sample
+        :param verbose: Bool of whether to print he TQDM bar or not.
+        :param start: Override the default start token.
+        :param delimiter: Override the default delimiter token.
+        :param classify: Override the default classify token
+        :return: Formatted outputs of the form. [batch, num_tokens'] where num_tokens' <= max_length
+        """
+        start = start or self.start
+        delimiter = delimiter or self.delimiter
+        clf_token = classify or self.clf_token
         num_samples = len(encoded)
         adjusted_max_length = max_length - num_samples - 1
         allocated_max_len = adjusted_max_length // num_samples
         outputs = []
-        for single_datum in zip(*encoded):
+        for single_datum in tqdm(zip(*encoded), disable=not verbose):
             overflows = [allocated_max_len - len(sequence) for sequence in single_datum]
             spare = sum(overflows)
             if spare >= 0:
                 cut_len = None
             else:
                 empty_tokens = sum(max(overflow, 0) for overflow in overflows)
                 num_over = [min(overflow, 0) for overflow in overflows].count(0)
-                cut_len = allocated_max_len + (empty_tokens // num_over)
-            joined = [self.start]
+                if num_over == 0:
+                    cut_len = allocated_max_len
+                else:
+                    cut_len = allocated_max_len + (empty_tokens // num_over)
+            joined = [start]
             for d in single_datum:
-                joined += (d[:cut_len] + [self.delimiter])
-            joined = joined[:-1] + [self.clf_token]
+                joined += (d[:cut_len] + [delimiter])
+            joined = joined[:-1] + [clf_token]
             outputs.append(joined)
         return outputs
+
+    def encode_input_sequence_labeling(self, *Xs, max_length, verbose=True):
+        # Xs = [n_inputs, n_batch, n_items_in_seq, 2]
+        text = []
+        labels_out = []
+        for input in Xs:
+            input_text = []
+            input_labels = []
+            for batch in input:
+                tokens, labels, _ = self._encode(batch)
+                tokens_single = functools.reduce(lambda x, y: x + y, tokens, [])
+                labels_single = functools.reduce(lambda x, y: x + y, labels, [])
+                input_text.append(tokens_single)
+                input_labels.append(labels_single)
+            text.append(input_text)
+            labels_out.append(input_labels)
+
+        return (self._cut_and_concat(text, max_length, verbose),
+                self._cut_and_concat(labels_out, max_length, verbose, PAD_LABEL, PAD_LABEL, PAD_LABEL))
```

Comment:
Use intermediate vars for storing result and call using kwargs rather than six positional args.